pkg/scenario/stage.go

```diff
@@ -166,6 +185,50 @@ func (c *StageClient) PlanWithSpecificVariableValueToExpect(t *testing.T, option
 	compareValues(t, actualValue, expectedValue, variable)
 }
 
+// PlanAndAssertJSONWithJSONPath performs JSON path planning and assertion in Go testing.
+//
+// t *testing.T: Testing object
+// options *terraform.Options: Terraform options
+// testCases []JSONPathTestCases: Array of JSON path test cases
+func (c *StageClient) PlanAndAssertJSONWithJSONPath(t *testing.T, options *terraform.Options, testCases []JSONPathTestCases) {
+	jsonPlan := terraform.InitAndPlanAndShow(t, options)
+
+	for _, testCase := range testCases {
+		t.Run(testCase.TestName, func(t *testing.T) {
+			var result []interface{}
+			k8s.UnmarshalJSONPath(t, []byte(jsonPlan), testCase.JSONPathToCompare, &result)
+			assert.NotNil(t, result)
+
+			// if expected is slice then it's ok to compare entire slice
+			if reflect.TypeOf(testCase.ExpectedValue).Kind() == reflect.Slice {
+				assert.ObjectsAreEqual(testCase.ExpectedValue, result)
+			} else {
+				// work on the result[0]
+				t.Logf("result returned raw: %v", result)
+				v := result[0]
+				if !testCase.AllowDifferentType {
+					assert.Equal(t, reflect.TypeOf(testCase.ExpectedValue).Kind(), reflect.TypeOf(v).Kind())
+				}
+				applyTestType(t, testCase.TestType, v, testCase.ExpectedValue, fmt.Sprintf("JSONPATH query: %s", testCase.JSONPathToCompare))
+			}
+		})
+	}
+}
+
+func applyTestType(t *testing.T, testType TestType, actual, expected interface{}, additionalMessage string) {
+	switch testType {
+	case ShouldContain:
+		msg := fmt.Sprintf("Output did not contains the expected value. Expected: %v, Actual: %v, %s", expected, actual, additionalMessage)
+		assert.Contains(t, actual, expected, msg)
+	case ShouldNOTContain:
+		msg := fmt.Sprintf("Output is expected NOT to contains the value. Expected: %v, Actual: %v, %s", expected, actual, additionalMessage)
+		assert.NotContains(t, actual, expected, msg)
+	case ShouldBeEqual:
+		msg := fmt.Sprintf("Output did not match with expected value. Expected: %v, Actual: %v, %s", expected, actual, additionalMessage)
+		assert.EqualValuesf(t, actual, expected, msg)
+	}
+}
+
 func compareValues(t *testing.T, actual interface{}, expected, variableName string) {
```

Comment:
Curious, why require and not assert? I've noticed that in the applyTestType you're proposing assert, yet here require. If require doesn't pass, it's going to stop the whole test execution - is what you're looking to accomplish?

Reply:
I'm using assert instead require for that specific reaosn. I would like to have a global summary with what passed and what not, so `assert` seems the proper way to do it. But perhaps I'm not getting the point of your comment.